src/utils/utils.rs

```diff
@@ -522,11 +510,12 @@ fn dot_dir(name: &str) -> Option<PathBuf> {
 }
 
 pub fn rustup_home_in_user_dir() -> Result<PathBuf> {
-    dot_dir(".rustup").ok_or_else(|| ErrorKind::RustupHome.into())
+    // XXX: This error message seems wrong/bogus.
+    dot_dir(".rustup").ok_or_else(|| anyhow::anyhow!("couldn't find value of RUSTUP_HOME"))
```

Comment:
I suppose the question here is how could `home_dir()` fail since that's the problem here.  Perhaps:
```suggestion
    dot_dir(".rustup").ok_or_else(|| anyhow::anyhow!("couldn't find home directory to construct RUSTUP_HOME"))
```
?

Reply:
RUSTUP_HOME might be unset
HOME might be unset
the users SID might not have a home dir set in the registry properly
....

Reply:
the patch just inlines the message for now, and the XXX is me whinging about our qualitay.

Reply:
Okay, this can sit out for now.